src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -2,70 +2,101 @@
+ * <li> Must have a type that is a subclass of the plugin base class returned by the {@link #getPluginBaseClass()}
+ *      method </li>
+ * <li> Must have a unique simple class name in order to avoid command line name collisions </li>
+ * <li> Should contain @Argument annotated fields for any arguments used with that plugin. Plugin arguments
+ *      may be optional or required. If required, the arguments are only "conditionally required", in that they
+ *      are required contingent on the predecessor plugin being specified on the command line, or included in the
+ *      default plugins (as determined by the command line parser via a call to isDependentArgumentAllowed).
+ *      Likewise, the same conditions must apply for any such dependent argument to be accepted as valid by the
+ *      parser.</li>
+ * <li> NOTE: plugin class @Arguments that are marked "optional=false" should not have a primitive type, and
+ *      should not have an initial value, as the command line parser will interpret these as having been set even
+ *      if they have not been specified on the command line. Conversely, @Arguments that are optional=true should
+ *      have an initial value, since the parser will not require them to be set in the command line. </li>
+ * </ul>
+ * <p>
+ * The command line parser calls the descriptor methods based on the following descriptor lifecycle:
+ * </p>
+ * <ol>
+ * <li> Durint command line parser initialization (before plugin discovery or command line parsing begins): </li>
+ * <ul>
+ * <li> {@link #getPackageNames} </li>
+ * <li> {@link #getPluginBaseClass} </li>
+ * </ul>
+ * <li> During plugin discovery (before argument parsing begins). Once for each plugin class discovered: </li>
+ * <ul>
+ * <li> {@link #includePluginClass} </li>
+ * <li> {@link #createInstanceForPlugin} </li>
+ * </ul>
+ * <li> During command line argument parsing: </li>
+ * <ul>
+ * <li> {@link #isDependentArgumentAllowed} - once for each argument that has been seen on the command line that is
+ *      contained within an plugin class that is controlled by this descriptor. The descriptor should decide whether
+ *      the dependent is allowed argument by determining if the predecessor class is valid, usually based on what
+ *      plugin name arguments and default instances have been provided for this descriptor.
+ * </li>
+ * </ul>
+ * <li> When command line argument parsing is complete (all command line arguments have been processed), but
+ * before the parser returns control to the caller, a one time call: </li>
+ * <ul>
+ * <li> {@link #validateAndResolvePlugins} </li>
+ * </ul>
+ * <li> After the command line parser returns from argument parsing. These methods are generally called by
+ * the application that consumes the plugins: </li>
+ * <ul>
+ * <li> {@link #getDefaultInstances} - whenever a plugin consumer wants the list of plugin instances that were
+ *      specified as defaults </li>
```

Comment:
As specified in the javadoc for this method, this is dependent on the resolution policy too, so it won't always return the list of plugins specified as defaults. Maybe a small note here will be useful.

Reply:
Done.